version.mk

```diff
@@ -0,0 +1,20 @@
+# Set version variables for LDFLAGS
```

Comment:
sorry about all this shell nonsense. this is slightly modified version of https://github.com/sigstore/cosign/blob/main/Makefile#L26-L43

Reply:
Can we move this logic to `Makefile`?

Also, why a modified version? What was different here compared to cosign?

Reply:
Didn't want to crowd the makefile, but I can move it.

1. so `PKG` is now calculated using `go list -m`, this allows for automatic package detection if the module changes from v2 -> v3 or whatever.
1. "clean" -> clean, "dirty" -> dirty, these seems unnecessarily quoted
1. VERSION_FLAGS is not quoted, because it needs to be appended to other ldflag options

Reply:
Makes sense. Thanks for explaining.